src/Compilers/CSharp/Portable/Symbols/NamespaceSymbol.cs

```diff
@@ -311,6 +311,23 @@ internal NamespaceSymbol GetNestedNamespace(NameSyntax name)
             return null;
         }
 
+        private ImmutableArray<NamedTypeSymbol> _lazyTypesWithExtensionMethods;
+        private ImmutableArray<NamedTypeSymbol> TypesWithExtensionMethods
+        {
+            get
+            {
+                var typesWithExtensionMethods = this._lazyTypesWithExtensionMethods;
+                if (typesWithExtensionMethods.IsDefault)
+                {
+                    this._lazyTypesWithExtensionMethods = this.GetTypeMembersUnordered().WhereAsArray(t => t.MightContainExtensionMethods);
```

Comment:
This does not need to be interlocked. There is no dependency on the list identity, so regular assignment is good enough.